File: java-automation/src/main/java/com/automation/pages/ProductDetailPage.java
package com.automation.pages;

import com.microsoft.playwright.Page;
import com.microsoft.playwright.Locator;
import com.automation.data.TestData;

import static org.junit.jupiter.api.Assertions.*;

public class ProductDetailPage {
    private final Page page;

    public ProductDetailPage(Page page) {
        this.page = page;
    }

    private Locator getProductName() {
        return page.locator(".inventory_details_name");
    }

    private Locator getProductPrice() {
        return page.locator(".inventory_details_price");
    }

    private Locator getProductDescription() {
        return page.locator(".inventory_details_desc");
    }

    private Locator getAddToCartButton() {
        return page.locator("[data-test='add-to-cart']");
    }

    private Locator getBackButton() {
        return page.locator("[data-test='back-to-products']");
    }

    public void assertProductDetail(String expectedName, String expectedPrice) {
        assertEquals(expectedName, getProductName().textContent(), 
            "Product name should match");
        assertEquals(expectedPrice, getProductPrice().textContent(), 
            "Product price should match");
        assertTrue(getProductDescription().isVisible(), 
            "Product description should be visible");
    }

    public void addToCart() {
        getAddToCartButton().click();
    }

    public void backToProducts() {
        getBackButton().click();
        assertEquals(TestData.Urls.INVENTORY, page.url(), 
            "User should be redirected to inventory page");
    }
}

File: python-automation/src/automation/pages/product_detail_page.py
from playwright.sync_api import Page
from src.automation.data.test_data import Urls


class ProductDetailPage:
    def __init__(self, page: Page):
        self.page = page

    @property
    def product_name(self):
        return self.page.locator(".inventory_details_name")

    @property
    def product_price(self):
        return self.page.locator(".inventory_details_price")

    @property
    def product_description(self):
        return self.page.locator(".inventory_details_desc")

    @property
    def add_to_cart_button(self):
        return self.page.locator("[data-test='add-to-cart']")

    @property
    def back_button(self):
        return self.page.locator("[data-test='back-to-products']")

    def assert_product_detail(self, expected_name: str, expected_price: str):
        actual_name = self.product_name.text_content()
        actual_price = self.product_price.text_content()

        assert actual_name == expected_name, \
            f"Product name should be '{expected_name}' but got '{actual_name}'"
        assert actual_price == expected_price, \
            f"Product price should be '{expected_price}' but got '{actual_price}'"
        assert self.product_description.is_visible(), \
            "Product description should be visible"

    def add_to_cart(self):
        self.add_to_cart_button.click()

    def back_to_products(self):
        self.back_button.click()
        assert self.page.url == Urls.INVENTORY, \
            f"User should be redirected to {Urls.INVENTORY}, but got {self.page.url}"
